src/tbx/mod.rs

```diff
@@ -151,16 +151,22 @@ impl Reader {
         let path = ffi::CString::new(path).unwrap();
         let c_str = ffi::CString::new("r").unwrap();
         let hts_file = unsafe { htslib::hts_open(path.as_ptr(), c_str.as_ptr()) };
-        unsafe {
-            if (*hts_file).format.category != htslib::htsFormatCategory_region_list
-                && (*hts_file).format.format != htslib::htsExactFormat_sam
-            {
-                return Err(Error::InvalidIndex);
-            }
+        let file_format = unsafe { (*hts_file).format.format };
+        let format_category = unsafe { (*hts_file).format.category };
+        match (format_category, file_format) {
+            (htslib::htsFormatCategory_region_list, _) => (),
+            (_, htslib::htsExactFormat_sam) => (),
```

Comment:
I wonder if this line permitting "sam" format is no longer needed, based on the htslib changelog here:
https://github.com/samtools/htslib/blob/82d01ba63b20eed8905ea60c2e1cfe0f96c73bab/NEWS#L222
The tests pass if it is removed; is that sufficient to conclude it's no longer needed?

Reply:
Discussion on #194 confirms that this interpretation is correct, and also suggests that this entire check may be unnecessary.